`string` cannot be used for `PopulatedDoc<T, ObjectId>` (regression)

### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

since 9.3.0 (tested in 9.6.0)

### Node.js version

any

### MongoDB server version

unrelated

### Typescript version (if applicable)

5.9.3

### Description

Since 9.3.0 (bisect says since 3971dc72d562810b6f1287b17a7289c5b16d4e71) using `mongoose.PopulatedDoc` with `ObjectId` as the RefType (default), it is not possible to assign `string` to it, resulting in:

```text
src/test.ts:30:43 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Type 'string' is not assignable to type 'ObjectId | { dummy?: string | undefined; } | { [x: string]: unknown; readonly _bsontype?: RegExp | "ObjectId" | undefined; id?: { [x: number]: number | undefined; readonly BYTES_PER_ELEMENT?: number | undefined; ... 37 more ...; readonly [Symbol.toStringTag]?: "Uint8Array" | undefined; } | { ...; } | undefined; ... ...'.

30   const other = await otherModel.create({ to: userId });
                                             ~~

  src/test.ts:12:3
    12   to: mongoose.PopulatedDoc<User>;
         ~~
    The expected type comes from property 'to' which is declared here on type '{ to?: ObjectId | { dummy?: string | undefined; } | { [x: string]: unknown; readonly _bsontype?: RegExp | "ObjectId" | undefined; id?: { [x: number]: number | undefined; readonly BYTES_PER_ELEMENT?: number | undefined; ... 37 more ...; readonly [Symbol.toStringTag]?: "Uint8Array" | undefined; } | { ...; } | undefine...'
  node_modules/mongoose/types/models.d.ts:315:5
    315     create(...docs: Array<DeepPartial<ApplyBasicCreateCasting<Require_id<TRawDocType>>>>): Promise<THydratedDocumentType[]>;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The last overload is declared here.


Found 1 error in src/test.ts:30
```

Note that this does *not* happen if the type is changed to just `mongoose.Types.ObjectId` or when using schema inference.

### Steps to Reproduce

```ts
// NodeJS: 25.2.1
// MongoDB: 7.0 (Docker)
// Typescript 5.9.3
import * as mongoose from 'mongoose'; // mongoose@9.6.0

interface User {
  dummy: string;
}

interface Other {
  to: mongoose.PopulatedDoc<User>;
}

const userSchema = new mongoose.Schema({ dummy: String });
const userModel = mongoose.model<User>('user', userSchema);

const otherSchema = new mongoose.Schema({ to: { type: mongoose.Types.ObjectId, ref: 'user' } });
const otherModel = mongoose.model<Other>('other', otherSchema);

async function main() {
  await mongoose.connect(`mongodb://localhost:27017/`, {
    dbName: 'verifyMASTER',
  });

  const user = await userModel.create({ dummy: 'hello' });

  const userId = user._id.toString();

  // type error: "Type 'string' is not assignable to ..."
  const other = await otherModel.create({ to: userId });

  console.log('Done', other);

  await mongoose.disconnect();
}

main();
```

[Reproduction repository & branch](https://github.com/typegoose/typegoose-testing/tree/discordE1ecti)

### Expected Behavior

To also allow `string` as a possibility when `ObjectId` is used.

--- a/test/types/create.test.ts
+++ b/test/types/create.test.ts
@@ -208,6 +208,27 @@ async function createWithRawDocTypeNo_id() {
   expect(doc2._id).type.toBe<Types.ObjectId>();
 }
 
+async function createWithPopulatedDoc() {
+  interface User {
+    dummy: string;
+  }
+
+  interface Other {
+    to: mongoose.PopulatedDoc<User>;
+  }
+
+  const userSchema = new mongoose.Schema({ dummy: String });
+  const userModel = mongoose.model<User>('TestCreateWithPopulatedDocUser', userSchema);
+
+  const otherSchema = new mongoose.Schema({ to: { type: mongoose.Types.ObjectId, ref: 'TestCreateWithPopulatedDocUser' } });
+  const otherModel = mongoose.model<Other>('TestCreateWithPopulatedDocOther', otherSchema);
+
+  const user = await userModel.create({ dummy: 'hello' });
+  const other = await otherModel.create({ to: user._id.toString() });
+
+  expect(other.to).type.toBe<mongoose.PopulatedDoc<User>>();
+}
+
 createWithAggregateErrors();
 
 async function gh15902() {

--- a/types/models.d.ts
+++ b/types/models.d.ts
@@ -227,15 +227,15 @@ declare module 'mongoose' {
   type CreateObjectWithExtraKeys<T> = T & Record<string, unknown>;
   type ApplyBasicCreateCasting<T> = {
     [K in keyof T]: NonNullable<T[K]> extends Map<infer KeyType extends string, infer ValueType>
-      ? (Record<KeyType, ValueType> | Array<[KeyType, ValueType]> | T[K])
+      ? (Record<KeyType, ValueType> | Array<[KeyType, ValueType]> | T[K] | QueryTypeCasting<Extract<T[K], TreatAsPrimitives>>)
       : NonNullable<T[K]> extends Types.DocumentArray<infer RawSubdocType>
-         ? RawSubdocType[] | T[K]
+         ? RawSubdocType[] | T[K] | QueryTypeCasting<Extract<T[K], TreatAsPrimitives>>
          : NonNullable<T[K]> extends Document<any, any, infer RawSubdocType>
-           ? CreateObjectWithExtraKeys<ApplyBasicCreateCasting<RawSubdocType>> | T[K]
+           ? CreateObjectWithExtraKeys<ApplyBasicCreateCasting<RawSubdocType>> | T[K] | QueryTypeCasting<Extract<T[K], TreatAsPrimitives>>
            : NonNullable<T[K]> extends TreatAsPrimitives
               ? QueryTypeCasting<T[K]>
               : NonNullable<T[K]> extends object
-                ? CreateObjectWithExtraKeys<ApplyBasicCreateCasting<T[K]>> | T[K]
+                ? CreateObjectWithExtraKeys<ApplyBasicCreateCasting<T[K]>> | T[K] | QueryTypeCasting<Extract<T[K], TreatAsPrimitives>>
                 : QueryTypeCasting<T[K]>;
   };
 

